components/sidebar/sidebar_category/sidebar_category_sorting_menu/sidebar_category_sorting_menu.tsx

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {injectIntl, IntlShape} from 'react-intl';
+
+import {Preferences} from 'mattermost-redux/constants';
+import {ChannelCategory, CategorySorting} from 'mattermost-redux/types/channel_categories';
+import {PreferenceType} from 'mattermost-redux/types/preferences';
+
+import Constants from 'utils/constants';
+
+import {trackEvent} from 'actions/telemetry_actions';
+
+import SidebarMenu from 'components/sidebar/sidebar_menu';
+import SidebarMenuType from 'components/sidebar/sidebar_menu/sidebar_menu';
+import Menu from 'components/widgets/menu/menu';
+import {Props as SubmenuItemProps} from 'components/widgets/menu/menu_items/submenu_item';
+
+type Props = {
+    category: ChannelCategory;
+    handleOpenDirectMessagesModal: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    categories?: ChannelCategory[];
+    intl: IntlShape;
+    isCollapsed: boolean;
+    isMenuOpen: boolean;
+    onToggleMenu: (isMenuOpen: boolean) => void;
+    currentUserId: string;
+    actions: {
+        setCategorySorting: (categoryId: string, sorting: CategorySorting) => void;
+        savePreferences: (userId: string, preferences: PreferenceType[]) => void;
+    };
+};
+
+type State = {
+    openUp: boolean;
+    width: number;
+    selectedDmNumber: number;
+    selectedCategorySort: string;
```

Comment:
The DM count needs to come from preferences, and the sorting method needs to come from the category itself still.